src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
```

Comment:
separate entites you would like to use for naming, better name if EMPTY_ARRAY_DATA